app/bot/broadcast_status.go

```diff
@@ -0,0 +1,115 @@
+package bot
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+)
+
+const (
+	MsgBroadcastStarted  = "Вещание началось"
+	MsgBroadcastFinished = "Вещание завершилось"
+)
+
+// BroadcastStatus bot returns current broadcast status
+type BroadcastStatus struct {
+	// Params
+	broadcastUrl string        // Url for "ping"
+	pingInterval time.Duration // Ping interval
+	delayToOff   time.Duration // After this interval of not OK, status will be switcher to OFF
+
+	// State
+	status         bool  // current broadcast status
+	lastSentStatus *bool // by default this value is null to react on first message
+	offPeriod      time.Duration
+	lastCheck      time.Time
+
+	// ping func (for tests)
+	ping func(ctx context.Context, url string) bool
+}
+
+// NewBroadcastStatus starts status checking goroutine and returns bot instance
+func NewBroadcastStatus(ctx context.Context, broadcastUrl string, pingInterval time.Duration, delayToOff time.Duration) *BroadcastStatus {
```

Comment:
it accepts 4 params already and I just suggested to add one more for http client. Wrapping them into a struct, i.e. BroadcastParams will be more readable and the call will have just two - ctx and BroadcastParams

Reply:
Done, but now we have BroadcastParams with 4 params :)

Reply:
well, this was the goal - named parameters via struct